src/utils/who.is.making.filling.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from "express";
+
+import { logger } from "./logger";
+import * as config from "../config";
+import { ApplicationData } from "../model";
+import { getApplicationData, setExtraData } from "./application.data";
+import { WhoIsRegisteringKey, WhoIsRegisteringType } from "../model/who.is.making.filing.model";
+
+export const getWhoIsFilling = (req: Request, res: Response, next: NextFunction, templateName: string, backLinkUrl: string): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+    const appData: ApplicationData = getApplicationData(req.session);
+
+    return res.render(templateName, {
+      backLinkUrl: backLinkUrl,
+      templateName: templateName,
+      [WhoIsRegisteringKey]: appData[WhoIsRegisteringKey]
+    });
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+export const postWhoIsFilling = (req: Request, res: Response, next: NextFunction, registrationFlag: boolean) => {
+  try {
+    logger.debugRequest(req, `POST ${config.WHO_IS_MAKING_FILING_PAGE}`);
+    const whoIsRegistering = req.body[WhoIsRegisteringKey];
+
+    setExtraData(req.session, { ...getApplicationData(req.session), [WhoIsRegisteringKey]: whoIsRegistering });
+
+    filterJourneyType(res, registrationFlag, whoIsRegistering);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const filterJourneyType = (res: Response, registrationFlag: boolean, whoIsRegistering: string) => {
+  if (registrationFlag) {
+    return whoIsRegistering === WhoIsRegisteringType.AGENT ? res.redirect(config.DUE_DILIGENCE_URL) : res.redirect(config.OVERSEAS_ENTITY_DUE_DILIGENCE_URL);
+  } else {
+    return res.redirect(config.OVERSEAS_ENTITY_REVIEW_PAGE);
```

Comment:
Does this not need updating to branch now?

Reply:
Ah just realising @olaych has added it as a ternary operator on the previous line so this else is remains of a merge or something. Will remove

Reply:
I'm wrong, this code block is used by both register and update journeys so the else block is for register, is that right @olaych ?

Reply:
else is for update. Registration branched on its own

Reply:
resolved this now @AndreKramer7654 , passing in the destination URLs to the shared block of code from each controller